sled-agent/src/rack_setup/service.rs

```diff
@@ -988,6 +1006,119 @@ impl ServiceInner {
         // DNS configuration to the internal DNS servers.
         self.initialize_dns(&service_plan).await?;
 
+        // Initialize rack network before NTP comes online, otherwise boundary
+        // services will not be available and NTP will fail to sync
+        info!(self.log, "Checking for Rack Network Configuration");
+        if let Some(rack_network_config) = &config.rack_network_config {
+            info!(self.log, "Initializing Rack Network");
+            info!(self.log, "Looking up address for Dendrite");
+            let resolver = DnsResolver::new_from_subnet(
+                self.log.new(o!("component" => "DnsResolver")),
+                config.az_subnet(),
+            )
+            .expect("Failed to create DNS resolver");
```

Comment:
Initially wasn't sure about reaching for `expect` here. But then I traced the `new_from_subnet` call down to a [`new_with_conn`](https://docs.rs/trust-dns-resolver/latest/src/trust_dns_resolver/async_resolver.rs.html#207) call, and it seems like this call cannot return an error? But I still wonder if we should return an error here instead of `expect`, so if the underlying implementation changes, we won't panic here.

Reply:
I yoinked this code from somewhere else and the `expect` came along for the ride 😂 
I'll handle the error appropriately, good catch.